### PR TITLE
Resolve tunnel service ports via RSD info fetched from given tunnel (iOS17)

### PR DESCRIPTION
**Pull request by diegoperini**

### CODE REVIEW

File: ios/connect.go [hunk 1]
```diff
@@ -126,7 +126,7 @@ func ConnectToServiceTunnelIface(device DeviceEntry, serviceName string) (Device
 }
 
 func ConnectToHttp2(device DeviceEntry, port int) (*http.HttpConnection, error) {
-	addr, err := net.ResolveTCPAddr("tcp6", fmt.Sprintf("[%s]:%d", device.Address, port))
+	addr, err := net.ResolveTCPAddr("tcp6", fmt.Sprintf("[%s]:%d", device.RsdAddress, port))
```
Comment (dmissmann) [inline ios/connect.go:129]: It is the address of the device, RSD has just it's own port like all other services.

File: ios/listdevices.go [hunk 2]
```diff
@@ -56,7 +56,8 @@ type DeviceEntry struct {
 	DeviceID         int
 	MessageType      string
 	Properties       DeviceProperties
-	Address          string
+	RsdAddress       string
+	RsdPort          int
```
Comment (dmissmann) [inline ios/listdevices.go:60]: `RsdPort` is never used

File: main.go [hunk 3]
```diff
@@ -130,14 +130,16 @@ Usage:
...
-  --udid=<udid>  		UDID of the device.
-  --address=<ipv6addrr>	Address of the device interface
-  --rsd=<path>			Path to RSD info
+  -v --verbose   		    Enable Debug Logging.
+  -t --trace     		    Enable Trace Logging (dump every message).
+  --nojson       		    Disable JSON output
+  --pretty       		    Pretty-print JSON command output
+  -h --help      		    Show this screen.
+  --udid=<udid>  		    UDID of the device.
+  --rsd=<path>			    Path to RSD info
```
Comment (dmissmann) [inline main.go:139]: this should be obsolete now

File: main.go [hunk 4]
```diff
@@ -1951,6 +1958,17 @@ func startTunnel(device ios.DeviceEntry) {
 	exitIfError("", err)
 }
 
+func updateDeviceRsd(device ios.DeviceEntry, udid string, rsdAddress string, rsdPort int) ios.DeviceEntry {
```
Comment (dmissmann) [inline main.go:1961]:
```suggestion
func deviceWithRsdProvider(device ios.DeviceEntry, udid string, rsdAddress string, rsdPort int) ios.DeviceEntry {
```

Comment (diegoperini) [reply main.go]: Applied.

File: main.go (hunk 3 above)
Comment (diegoperini) [reply main.go]: Removed.

File: ios/listdevices.go (hunk 2 above)
Comment (diegoperini) [reply ios/listdevices.go]: Removed

File: ios/connect.go (hunk 1 above)
Comment (diegoperini) [reply ios/connect.go]: Renamed.

**Review by dmissmann — APPROVED** [review]

**Pull request merged**